mailpoet/lib/Twig/Analytics.php

```diff
@@ -9,32 +9,59 @@
 use MailPoetVendor\Twig\TwigFunction;
 
 class Analytics extends AbstractExtension {
+
+  /** @var  AnalyticsGenerator */
+  private $analytics;
+
   public function getFunctions() {
-    $analytics = ContainerWrapper::getInstance()->get(AnalyticsGenerator::class);
-    if (!$analytics instanceof AnalyticsGenerator) {
-      throw new InvalidStateException('AnalyticsGenerator service was not registered!');
-    }
     return [
       new TwigFunction(
         'get_analytics_data',
-        [$analytics, 'generateAnalytics'],
+        [$this, 'generateAnalytics'],
         ['is_safe' => ['all']]
       ),
       new TwigFunction(
         'is_analytics_enabled',
-        [$analytics, 'isEnabled'],
+        [$this, 'isEnabled'],
         ['is_safe' => ['all']]
       ),
       new TwigFunction(
         'get_analytics_public_id',
-        [$analytics, 'getPublicId'],
+        [$this, 'getPublicId'],
         ['is_safe' => ['all']]
       ),
       new TwigFunction(
         'is_analytics_public_id_new',
-        [$analytics, 'isPublicIdNew'],
+        [$this, 'isPublicIdNew'],
         ['is_safe' => ['all']]
       ),
     ];
   }
+
+  private function analytics() {
```

Comment:
For a more descriptive name and a better consistency, I'd maybe call this `getAnalytics`.

Reply:
[0665af1](https://github.com/mailpoet/mailpoet/pull/4149/commits/0665af1dc24c78f7edc471f41941172fdc329abd)